wn_load.pl

```diff
@@ -13,34 +13,36 @@
 
 allwn(L):-
   semrels(L),
-  writef('\nSemantic Relations: %w\n', [L]).
+  format('\nSemantic Relations: ~w\n', [L]).
 allwn(L):-
   lexrels(L),
-  writef('\nLexical Relations: %w\n', [L]).
+  format('\nLexical Relations: ~w\n', [L]).
 allwn(L):-
   lexinfo(L),
-  writef('\nLexical Info: %w\n', [L]).
+  format('\nLexical Info: ~w\n', [L]).
 allwn(L):-
   seminfo(L),
-  writef('\nSemantic Info: %w\n', [L]).
+  format('\nSemantic Info: ~w\n', [L]).
 allwn(L):-
   morphinfo(L),
-  writef('\nMorphological Info: %w\n', [L]).
+  format('\nMorphological Info: ~w\n', [L]).
 
 /* ------------------------------------------
 Load WN
 ------------------------------------------ */
 
+pred2arity(P,A):-
```

Comment:
current_predicate(P/A)  is already deterministic by definition, so the cut + extra pred2arity/2 wrapper isn’t needed.

Defining pred2arity like this should be enough:

```
pred2arity(P,A,L):-
  current_predicate(P/A),
  length(L,A).
```

Reply:
Supposing a wn version had predicates with the same name and different arities, then we would not want to discard them: they might be either legitimate, or errors that would need fixing. For ex: some versions of SWI carried "ghost" definitions for other arities of the  "s" predicate like s/1), which were not from Wordnet.